### ISSUE TITLE
[Bug]: File upload error issue when changing PDF_EXTRACT_IMAGES=true

### ISSUE THREAD

**Issue by architectyou** [danny-avila/rag_api#88]
### What happened?

I wanted to use the extract_images feature of the PyPDFLoader library, so I set up a variable in the .env file and set it to true. However, the document upload that was working fine before now produces an error. When I set it to false, the upload and RAG work well without any issues.

![image](https://github.com/user-attachments/assets/bcc48c1d-5ee0-484c-be1b-31b7e8ed138d)

However, when I run the main.py file of rag_api, it works fine without any problems. The PDF_EXTRACT_IMAGES variable is also recognized correctly.

Regardless of the file extension, errors occur for all files.

### Steps to Reproduce

Step 1: Initial Setup

To use the extract_images feature of the PyPDFLoader library, you added the PDF_EXTRACT_IMAGES variable to the .env file and set it to true.
Previously, document upload was working fine.

Step 2: Problem Occurrence

After setting PDF_EXTRACT_IMAGES to true, an error occurred during document upload.
Error message: "TypeError: 'NoneType' object is not subscriptable"
When PDF_EXTRACT_IMAGES is set to false, the upload and RAG work without any issues.

Step 3: Additional Testing

When running the main.py file of rag_api directly, it works fine without any problems.
In this case, the PDF_EXTRACT_IMAGES variable is recognized correctly.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by ScarFX** [comment]
Hi @architectyou , thanks for bringing this up. 

I am having trouble replicating the issue. Which .env file did you change, in rag_api? Libre chat?

**Comment by architectyou** [comment]
Thanks, @ScarFX . 

It seems the problem occurs while launching containers with Docker. When testing with the 'npm run backend' command and launching the RAG API with Docker (api-compose.yaml) to connect to LibreChat, file errors occur. The same happens when adding the RAG API to the 'docker-compose.yaml' in librechat.yaml.

However, no errors occur when only running main.py in the 'rag_api' folder and connecting it to the LibreChat backend. For 'npm run backend' testing, I modified the environment variables in the .env file within the RAG API folder. When building the entire LibreChat with Docker Compose, I modified the variables in LibreChat's .env file.

When I successfully ran only main.py, I used the .env code as shown above without modifications. It seems something gets tangled up while launching the containers.

Here are my ENV settings. The folder structure is the same as the original, except for the 'api-compose.yaml' volumes path


Here's my env settings.
the folder structure is same as original without 'api-compse.yaml' 's volumes path.

1. **rag_api .env**
![image](https://github.com/user-attachments/assets/f2dfcf5f-94f5-40cc-b66d-bdbb57b162e3)


2. **librechat .env**
![image](https://github.com/user-attachments/assets/b6095367-e9a3-4047-a701-3615f6a8469a)


3. **api-compose.yaml (env path)**
![image](https://github.com/user-attachments/assets/fc1cf15c-5d43-470f-a14a-457ba1cffb2f)


4. **librechat docker-compose.yaml (env path)**
![image](https://github.com/user-attachments/assets/a91469b7-7d01-435c-a6a0-40ffd4b760cf)

**Comment by ScarFX** [comment]
I partially was able to replicate your error messages. Running everything as docker containers and `PDF_EXTRACT_IMAGES=True` I got the error `rapidocr-onnxruntime package not found`. Only got the error (which is different from yours) when trying to upload pdfs with images, not with text only pdfs and other file types. 

To fix this, in the rag_api directory I changed the Dockerfile RUN command to 
```
RUN apt-get update \
    && apt-get install -y --no-install-recommends \
    pandoc \
    netcat-openbsd \
    libgl1-mesa-glx \  
    libglib2.0-0 \
    && rm -rf /var/lib/apt/lists/* 
```

![Dockerfile for rag_api](https://github.com/user-attachments/assets/258b638e-e91f-4ec6-9af9-46cffb95d01f)

@architectyou let me know if this fixes it or if you are having issues implementing this.

**Comment by ScarFX** [comment]
@architectyou Did it work? The rag_api image was updated Friday to reflect the change to the dockerfile I gave you.

**Comment by ScarFX** [comment]
@architectyou The docker image is updated. Did this solve your issue?

**Comment by architectyou** [comment]
@ScarFX No.. I haven't tested it yet. Still, Thank you for updating the Docker Images😅
